tensorflow/python/eager/forwardprop_test.py

```diff
@@ -246,6 +249,61 @@ def testJVPFunction(self):
         ))
     self.assertAllClose([2. * 5. + 3. * 4.], self.evaluate(vp))
 
+  def testJVPFunctionWithBatchOfTangents(self):
+    add_outputs = (constant_op.constant(4.),)
+    jvp_flat = forwardprop._jvp_dispatch(
+        op_name="Add",
+        attr_tuple=(),
+        inputs=(constant_op.constant(1.), constant_op.constant(3.)),
+        outputs=add_outputs,
+        tangents=(
+            constant_op.constant([1., 2., 3.]),
+            constant_op.constant([4., 5., 6.]),
+        ),
+        use_batch=True)
+
+    # Using evaluate and asserting with just a list works too
+    # but the output is more explicit this way
+    self.assertAllClose(
+        [constant_op.constant([1. + 4., 2. + 5., 3. + 6.])],
+        jvp_flat
+    )
+
+    mul_outputs = (constant_op.constant([20.]),)
+    jvp_flat = forwardprop._jvp_dispatch(
+        op_name="Mul",
+        attr_tuple=(),
+        inputs=(constant_op.constant([4.]), constant_op.constant([5.])),
+        outputs=mul_outputs,
+        tangents=(
+            constant_op.constant([[1.], [0.], [1.]]),
+            constant_op.constant([[0.], [1.], [1.]]),
+        ),
+        use_batch=True)
+    self.assertAllClose(
+        [constant_op.constant([[5.], [4.], [5. + 4.]])],
+        jvp_flat
+    )
+
+  def testJVPFunctionRaisesError(self):
+    context.ensure_initialized()
+    ctx = context.context()
```

Comment:
Removed. About the `gradient_impl` change, `parallel_for` also depends on `:test_util` which then leads to `python:gradients` again (test_util -> python/client_testlib -> python:gradient_checker). Anything that can be done here?

Reply:
The BUILD dependencies as you had them seemed fine (parallel_for:control_flow_ops rather than parallel_for:parallel_for). It was just the Python imports that were problematic because of the __init__ file. I made the one gradients -> gradients_impl change and it seemed fine. I think if we do that and you push a commit with the context changes we'll be good to commit this.

Reply:
Oh sure. Will make changes.